src/auth/index.ts

```diff
@@ -98,7 +98,7 @@ export default function createShopifyAuth(options: OAuthStartOptions) {
         ctx.state.shopify = await Shopify.Auth.validateAuthCallback(
           ctx.req,
           ctx.res,
-          ctx.query,
+          ctx.query as unknown as AuthQuery,
```

Comment:
Is the double cast necessary here?
```suggestion
          ctx.query as AuthQuery,
```

Reply:
Yes, because the type of `ctx.query` doesn't specify the fields as we do in `AuthQuery` (since we expect a specific set of fields). The alternative would be to create a new hash variable that cherry-picks the fields we require.